src/Api/PubnubApi/Pubnub.cs

```diff
@@ -42,10 +46,26 @@ public ISubscribeOperation<T> Subscribe<T>()
 		{
             if (pubnubConfig[InstanceId].EnableEventEngine)
             {
-                EndPoint.SubscribeOperation2<T> subscribeOperation = new EndPoint.SubscribeOperation2<T>(pubnubConfig.ContainsKey(InstanceId) ? pubnubConfig[InstanceId] : null, JsonPluggableLibrary, pubnubUnitTest, pubnubLog, null, tokenManager, this);
+                SubscribeEventEngine subscribeEventEngine;
+				Action<Pubnub, PNStatus> statusListener = null;
+				Action<Pubnub, PNMessageResult<T>> messageListener = null;
+				if (this.subscribeEventEngineFactory.hasEventEngine(InstanceId)) {
+                    subscribeEventEngine = subscribeEventEngineFactory.getEventEngine(InstanceId); }
+                else {
+					var subscribeManager = new SubscribeManager2(pubnubConfig[InstanceId], JsonPluggableLibrary, pubnubUnitTest, pubnubLog, telemetryManager, tokenManager, this);
+                    if (subscribeCallbackListenerList!= null && subscribeCallbackListenerList.Count > 0) {
+						messageListener = subscribeCallbackListenerList[0].Message;
+                        statusListener = subscribeCallbackListenerList[0].Status;
+					}
+                    subscribeEventEngine = subscribeEventEngineFactory.initializeEventEngine(InstanceId, this, pubnubConfig[InstanceId], subscribeManager, statusListener, messageListener);
+				}
+				EndPoint.SubscribeOperation2<T> subscribeOperation = new EndPoint.SubscribeOperation2<T>(pubnubConfig.ContainsKey(InstanceId) ? pubnubConfig[InstanceId] : null, JsonPluggableLibrary, pubnubUnitTest, pubnubLog, null, tokenManager, this);
                 subscribeOperation.SubscribeListenerList = subscribeCallbackListenerList;
-                //subscribeOperation.CurrentPubnubInstance(this);
-                savedSubscribeOperation = subscribeOperation;
+                
+
+				subscribeEventEngine.eventQueue.Enqueue(new SubscriptionChangedEvent() { Channels = this.GetSubscribedChannels().ToArray(), ChannelGroups = this.GetSubscribedChannelGroups().ToArray() });
```

Comment:
IMHO, Pubnub.cs is facade layer for all the operations that we want to expose to end users.

Reply:
WIP: finding more cleaner way!!!

Reply:
@budgetpreneur   If I expose `getSubscribedChannels()` in `SubscribeOperation2` then It will be easy to get that info for eventEngine. 
Because otherwise we need to pass something related to event engine into `SubscribeOperation2`.

Reply:
@budgetpreneur  please check last commit